### PR TITLE
fix(web): Only allow web to build if there are no TypeScript errors

### PR DESCRIPTION
**Pull request by RunarVestmann**
# Only allow web to build if there are no TypeScript errors

## What

* The web can be built even though there are type errors in the code
* In this PR the goal is to change settings so that in the future the web can't be built if there are type errors (for safety reasons) and also to address all currently existing type errors in the web project

Here is a PR for the NX project that was of interest https://github.com/nrwl/nx/pull/4970

What I've done is change the tsconfig.json file a bit. But perhaps there is a better way to actually make the web build fail due to build errors

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by thorkellmani — APPROVED** [review]

**Review by eirikurn — APPROVED** [review]
Pushed some more stuff.

We also need to upgrade the types in web to be "strict: true". It will catch more issues and reduce ugly type assertions (like `regulation.title as HTMLText`).

It looked like there were only 14 strict errors, which I fixed, but then it turned out that the error list is capped and there may be A LOT MORE. So let's work on that together in follow up PRs. 🙏

File: apps/web/components/ServiceWeb/Forms/StandardForm/StandardForm.tsx [hunk 1]
```diff
@@ -184,7 +184,9 @@ const BasicInput = ({
           },
         }),
       }}
-      {...register(name)}
+      // The docs tell us to spread the response of the register function even though it's return type is void
+      // https://react-hook-form.com/api/useformcontext/
+      {...((register(name) as unknown) as object)}
```
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply apps/web/components/ServiceWeb/Forms/StandardForm/StandardForm.tsx:189]: I believe team-juni wrote this code at the time so I'm not sure if this is done like this for some purpose 🤷

[the remark this answers is not in the document]
Comment (eirikurn) [reply apps/web/components/ServiceWeb/Forms/StandardForm/StandardForm.tsx:189]: Can you try removing it and see if the form still works?

**Review by siggifv — APPROVED** [review]

File: apps/web/components/ServiceWeb/Forms/StandardForm/StandardForm.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply apps/web/components/ServiceWeb/Forms/StandardForm/StandardForm.tsx:189]: I'll try that but I'm gonna put the automerge label on this PR :)

**Pull request merged**